plugins/maven-dependency-resolver/src/main/java/org/robolectric/internal/dependency/MavenJarArtifact.java

```diff
@@ -0,0 +1,50 @@
+package org.robolectric.internal.dependency;
+
+/**
+ * Encapsulates some parts of a Maven artifact. This assumes all artifacts are of type jar and do
+ * not have a classifier.
+ */
+public class MavenJarArtifact {
+  private final String groupId;
+  private final String artifactId;
+  private final String version;
+  private final String jarPath;
+  private final String jarSha1Path;
+  private final String pomPath;
+  private final String pomSha1Path;
+
+  public MavenJarArtifact(DependencyJar dependencyJar) {
+    this.groupId = dependencyJar.getGroupId();
+    this.artifactId = dependencyJar.getArtifactId();
+    this.version = dependencyJar.getVersion();
+    String basePath =
+        String.format("%s/%s/%s", this.groupId.replace(".", "/"), this.artifactId, this.version);
+    String baseName = String.format("%s-%s", this.artifactId, this.version);
+    this.jarPath = String.format("%s/%s.jar", basePath, baseName);
+    this.jarSha1Path = String.format("%s/%s.jar.sha1", basePath, baseName);
+    this.pomPath = String.format("%s/%s.pom", basePath, baseName);
+    this.pomSha1Path = String.format("%s/%s.pom.sha1", basePath, baseName);
```

Comment:
Nit: SHA1 is dubbed insecure and Gradle by default publishes SHA256 and SHA512 checksums to repositories that support it. I guess you've chosen it because it is the most compatible and currently still in use for the Android artifact publishing?

Reply:
For some reason Robolectric is not publishing SHA-256 and SHA-512 (only MD5 and SHA1).
https://repo1.maven.org/maven2/org/robolectric/robolectric/4.4-alpha-2/

Reply:
There was an attempt to switch to maven-publish in https://github.com/robolectric/robolectric/pull/4606/commits/60c0d0b0629819fdcf5179eaef0b98a383cc3517, but that got reverted. I think maven-publish does sha256 and sha512 by default.